app/Http/Controllers/Api/Remote/Backups/BackupRemoteUploadController.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+namespace Pterodactyl\Http\Controllers\Api\Remote\Backups;
+
+use Carbon\CarbonImmutable;
+use Illuminate\Http\Request;
+use Illuminate\Http\JsonResponse;
+use League\Flysystem\AwsS3v3\AwsS3Adapter;
+use Pterodactyl\Http\Controllers\Controller;
+use Pterodactyl\Extensions\Backups\BackupManager;
+use Pterodactyl\Repositories\Eloquent\BackupRepository;
+
+class BackupRemoteUploadController extends Controller
+{
+    const PART_SIZE = 5 * 1024 * 1024 * 1024;
+
+    /**
+     * @var \Pterodactyl\Repositories\Eloquent\BackupRepository
+     */
+    private $repository;
+
+    /**
+     * @var \Pterodactyl\Extensions\Backups\BackupManager
+     */
+    private $backupManager;
+
+    /**
+     * BackupRemoteUploadController constructor.
+     *
+     * @param \Pterodactyl\Repositories\Eloquent\BackupRepository $repository
+     * @param \Pterodactyl\Extensions\Backups\BackupManager $backupManager
+     */
+    public function __construct(BackupRepository $repository, BackupManager $backupManager)
+    {
+        $this->repository = $repository;
+        $this->backupManager = $backupManager;
+    }
+
+    /**
+     * ?
+     *
+     * @param \Illuminate\Http\Request $request
+     * @param string $backup
+     *
+     * @return \Illuminate\Http\JsonResponse
+     *
+     * @throws \Pterodactyl\Exceptions\Repository\RecordNotFoundException
+     * @throws \Exception
+     */
+    public function __invoke(Request $request, string $backup)
+    {
+        $size = $request->query('size', null);
+        if ($size === null) {
+            return new JsonResponse([], JsonResponse::HTTP_BAD_REQUEST);
```

Comment:
This should probably throw a `BadRequestHttpException` or `DisplayException` or something, try to avoid error responses with no details.